modules/kernel/test/org/apache/axis2/transport/http/util/XMLDeclarationTest.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.axis2.transport.http.util;
+
+import junit.framework.TestCase;
+import org.apache.axiom.om.OMAbstractFactory;
+import org.apache.axiom.om.OMOutputFormat;
+import org.apache.axiom.om.impl.OMNamespaceImpl;
+import org.apache.axiom.om.impl.llom.OMElementImpl;
+import org.apache.axiom.om.impl.llom.OMTextImpl;
+import org.apache.axiom.soap.SOAPEnvelope;
+import org.apache.axiom.soap.SOAPFactory;
+import org.apache.axis2.AxisFault;
+import org.apache.axis2.context.MessageContext;
+import org.apache.axis2.transport.http.ApplicationXMLFormatter;
+
+import java.io.ByteArrayOutputStream;
+import java.nio.charset.StandardCharsets;
+
+public class XMLDeclarationTest extends TestCase {
+
+    public void testWithoutDeclaration() throws Exception {
+        ApplicationXMLFormatter xmlFormatter = new ApplicationXMLFormatter();
+        MessageContext messageContext = new MessageContext();
+
+        createPayload(messageContext);
+
+        OMOutputFormat format = new OMOutputFormat();
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+
+        xmlFormatter.writeTo(messageContext, format, baos, true);
+        String output = new String(baos.toByteArray());
+        assertEquals("<hello>world</hello>", output);
+    }
+
+    public void testWithDeclaration() throws Exception {
+        ApplicationXMLFormatter xmlFormatter = new ApplicationXMLFormatter();
+        MessageContext messageContext = new MessageContext();
+        messageContext.setProperty("WRITE_XML_DECLARATION", "true");
+
+        createPayload(messageContext);
+
+        OMOutputFormat format = new OMOutputFormat();
+        format.setCharSetEncoding(StandardCharsets.UTF_8.toString());
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+
+        xmlFormatter.writeTo(messageContext, format, baos, true);
+        String output = new String(baos.toByteArray());
+        assertEquals("<?xml version=\"1.0\" encoding=\"UTF-8\" ?><hello>world</hello>", output);
```

Comment:
Add assert message?

Reply:
In this case, without the assert message the failure output gives an idea about the failure, so refrained adding message.

Reply:
Yes, true. But the failure message will say something like expected true but found false message. Therefore, it is better to have a detailed assert message in the log. WDYT?